server/controller/routing.go

```diff
@@ -34,22 +34,16 @@ func NewRoutingController(routing types.RoutingAPI, store types.StoreAPI) routin
 func (c *routingCtlr) Publish(ctx context.Context, req *routingv1.PublishRequest) (*emptypb.Empty, error) {
 	routingLogger.Debug("Called routing controller's Publish method", "req", req)
 
-	ref := &corev1.RecordRef{
-		Cid: req.GetRecordCid(),
-	}
-
-	record, err := c.getRecord(ctx, ref)
-	if err != nil {
-		st := status.Convert(err)
-
-		return nil, status.Errorf(st.Code(), "failed to get record: %s", st.Message())
-	}
-
-	err = c.routing.Publish(ctx, ref, record)
-	if err != nil {
-		st := status.Convert(err)
-
-		return nil, status.Errorf(st.Code(), "failed to publish: %s", st.Message())
+	switch req.GetRequest().(type) {
+	case *routingv1.PublishRequest_RecordRefs:
+		if err := c.processRecordRefs(ctx, req.GetRecordRefs().GetRefs(), c.routing.Publish); err != nil {
+			return nil, err
+		}
+
+	case *routingv1.PublishRequest_Queries:
+	case *routingv1.PublishRequest_AllRecords:
+	default:
+		return nil, status.Errorf(codes.InvalidArgument, "invalid publish request")
```

Comment:
According to my previous comment, this unnecessarily complicates the Publish service, and I don't think Publish is entirely "secure" in this form.